src/commonMain/kotlin/dev/romainguy/kotlin/math/Vector.kt

```diff
@@ -787,6 +787,14 @@ inline fun mix(a: Float4, b: Float4, x: Float4): Float4 {
             mix(a.w, b.w, x.w))
 }
 
+inline fun combine(a: Float4, b: Float4): Float4 {
```

Comment:
What does `combine` do? Does it compose quaternions? If so you'd need a better name. Or maybe use `typealias` so it's obvious when a `Float4` is a quaternion.

Reply:
>What does combine do? Does it compose quaternions?

Yes, `combine` composes 2 quaternions.
![image](https://user-images.githubusercontent.com/6597529/150168297-d572b1f1-660e-41bc-b0b4-e4556d08c88c.png)
*[Wikipedia - Quaternions and spatial rotation](https://en.wikipedia.org/wiki/Quaternions_and_spatial_rotation)*

>If so you'd need a better name

Actually, it should be put in an `operator fun Quaternion.times(q: Quaternion) :  Quaternion` but that would mean we need a `Quaternion` class since `Float4.times` already exist and since a `typealias` cannot have a dedicated fun.
So maybe I can call it `multiply`

>Or maybe use typealias so it's obvious when a Float4 is a quaternion.

I already typed aliased everything in SceneView like this:
```kotlin
typealias Transform = Mat4
typealias Position = Float3
typealias Rotation = Float3
typealias Scale = Float3
typealias Orientation = Float3
typealias Quaternion = Float4
```
Maybe I could make a new PR after this one with those typealiases integrated.

Reply:
No, only quaternions should have a typealias. Maybe it should even be its own type like you suggested. The other typealiases should not be in this library.

Reply:
Ok for the `Quaternion` own class but that will mean a big copy/paste/clean from `Float4` since data class cannot be inherited.
We can add the `times` operator first and maybe introduce others later depending on Quaternions real needs.

Reply:
The only question that remains is whether you want the Quaternion to be Object or Function oriented.
In case of `Quaternion` dedicated `data class` some Float4 related functions won't be accessible.

The Functions way doesn't mandatory exclude a dedicated `Quaternion.kt` file and also doesn't exclude `Quaternion` `typealias` usage in functions arguments. Typealias doesn't forbid Float4 argument passing when Quaternion is needed but Quaternion extensions will also be accessible at Float4 level.

Reply:
It's not a problem to replicate the content of `Float4`. I also imagine that some of the `Float4` properties and methods are not necessary on quaternions. In Filament, quaternions are [their own type](https://github.com/google/filament/blob/main/libs/math/include/math/quat.h) and only reuse a few things from the base `float4` like add/subtract/negate/comparisons. Multiplication, etc. are specific to quaternions, so it's probably best to do it as a different type to avoid potential misuse.

Reply:
Fixed https://github.com/romainguy/kotlin-math/pull/33/commits/bd1b5bb48a4af27fdfdd145ea27760d83a1f08b5